src/stores/filingHistoryListStore.ts

```diff
@@ -0,0 +1,322 @@
+import { ApiFilingIF, CommentIF, DocumentIF, FilingHistoryListStateIF } from '@/interfaces'
+import { DateUtilities, EnumUtilities, LegalServices } from '@/services'
+import { FilingTypes } from '@bcrs-shared-components/enums'
+import { defineStore } from 'pinia'
+import { useBusinessStore } from './businessStore'
+import { useRootStore } from './rootStore'
+
+export const useFilingHistoryListStore = defineStore('filingHistoryList', {
+  state: (): FilingHistoryListStateIF => ({
+    addCommentDialog: false,
+    currentFiling: null,
+    downloadErrorDialog: false,
+    fileCorrectionDialog: false,
+    filings: [],
+    loadingAll: false,
+    loadCorrectionDialog: false,
+    loadingOne: false,
+    loadingOneIndex: -1,
+    panel: null
+  }),
+
+  getters: {
+    getCurrentFiling (state): ApiFilingIF {
+      return state.currentFiling
+    },
+
+    /** The list of filings from the API. */
+    getFilings (state): ApiFilingIF[] {
+      // return only the valid filings
+      return state.filings.filter(filing => {
+        // safety check for required fields
+        if (!filing.name || !filing.displayName || !filing.effectiveDate || !filing.submittedDate || !filing.status) {
+          console.log('Invalid filing =', filing) // eslint-disable-line no-console
+          return false
+        }
+        return true
+      })
+    },
+
+    /** The count of filings in the Filing History List. */
+    getHistoryCount (): number {
+      const filings = this.getFilings as ApiFilingIF[]
```

Comment:
Is the "as" cast needed, or is the type of filings correct without it?

Reply:
When we do not cast it, my IDE says that filings has a type of any:
![Screenshot (65)](https://github.com/bcgov/business-filings-ui/assets/122301442/71893ef6-fa07-4d00-92df-e732b0149f98)

For line 59, typescript was complaining although `getFilings` has a return type of `ApiFilingIF[]`:
![Screenshot (66)](https://github.com/bcgov/business-filings-ui/assets/122301442/a36cfe2b-e10a-40ea-920c-206897d5cdc0)

Reply:
OK, I confirm that you need the `as` here. I think it's because this object is written in "options API" (methods are separated by commas) instead of "class API".

Darn, I was hoping local types would be known.